### PR TITLE
fix: check env before running kafka test

### PR DESCRIPTION
**Pull request by WenyXu**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

check env before running kafka test

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.
- [x]  This PR does not require documentation updates.

### CODE REVIEW

File: src/log-store/src/kafka/util/record.rs [hunk 1]
```diff
@@ -295,11 +295,11 @@ mod tests {
 
     use common_base::readable_size::ReadableSize;
     use common_config::wal::KafkaConfig;
-    use rand::Rng;
+    use uuid::Uuid;
 
     use super::*;
     use crate::kafka::client_manager::ClientManager;
-
+    use crate::kafka::util::test_util::run_test_with_kafka_wal;
```
Comment (niebayes) [inline src/log-store/src/kafka/util/record.rs:302]:
```suggestion
    use crate::kafka::util::test_util::run_test_with_kafka_wal;
    
```

File: src/log-store/src/kafka/util/test_util.rs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+pub async fn run_test_with_kafka_wal<F>(test: F)
+where
+    F: FnOnce(Vec<String>) -> BoxFuture<'static, ()>,
+{
+    let endpoints = env::var("GT_KAFKA_ENDPOINTS").unwrap_or_default();
+    if endpoints.is_empty() {
+        warn!("The endpoints is empty, skipping the test");
+        return;
+    }
```
Comment (niebayes) [inline src/log-store/src/kafka/util/test_util.rs:28]:
```suggestion
    let Ok(endpoints) = env::var("GT_KAFKA_ENDPOINTS") else {
        warn!("The endpoints is empty, skipping the test");
        return;
    };
```

**Review by niebayes — APPROVED** [review]
LGTM

**Comment by sunng87** [comment]
There is a potential leak that we expect the test to fail but it doesn't run.
I think this is better done via cargo test/nextest filters: https://nexte.st/book/filter-expressions. When we want to skip the test we skip it explicitly.

And one step further, I still suggest kafka related code to be feature-gated because: 

1. it's an optional feature; 
2. it introduces an external dependency. 

The feature can be enabled by default but it allows us to turn off when it's not needed.

**Comment by WenyXu** [comment]
> I think this is better done via cargo test/nextest filters: https://nexte.st/book/filter-expressions. When we want to skip the test we skip it explicitly.

Let's make the next nightly release tests work as expected first.  And then moving kafka-related code under the feature gate.  Actually, our region migration also relies on remote WAL features(currently, it's Kafka WAL). If we move all kafka code under the feature gate, there may be some related works that need to be done (at least all region migration integration tests also need to move under the feature gate)

**Review by MichaelScofield — APPROVED** [review]

**Pull request merged**